src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -2250,7 +2263,7 @@ namespace NServiceBus.Sagas
         public interface Using<T, M> : NServiceBus.Sagas.IFinder
             where T : NServiceBus.IContainSagaData
         {
-            System.Threading.Tasks.Task<T> FindBy(M message, NServiceBus.Extensibility.ReadOnlyContextBag context);
+            System.Threading.Tasks.Task<T> FindBy(M message, NServiceBus.Persistence.TransactionalStorageSession storageSession, NServiceBus.Extensibility.ReadOnlyContextBag context);
```

Comment:
I like your idea @andreasohlund

On Thursday, October 22, 2015, Andreas Öhlund <notifications@github.com>
wrote:

> In
> src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt
> <https://github.com/Particular/NServiceBus/pull/3024#discussion_r42735193>
> :
>
> > @@ -2250,7 +2263,7 @@ namespace NServiceBus.Sagas
> >          public interface Using<T, M> : NServiceBus.Sagas.IFinder
> >              where T : NServiceBus.IContainSagaData
> >          {
> > -            System.Threading.Tasks.Task<T> FindBy(M message, NServiceBus.Extensibility.ReadOnlyContextBag context);
> > +            System.Threading.Tasks.Task<T> FindBy(M message, NServiceBus.Persistence.TransactionalStorageSession storageSession, NServiceBus.Extensibility.ReadOnlyContextBag context);
>
> Since we now have both the session and the context should we introduce a
> explicit SagaFinderContext instead?
>
> So instead of
>
> MyFinder
> storageSession.GetRavenSession().Find(...)
>
> we get
>
> MyFinder
> finderContext.GetRavenSession().Find(...)
>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3024/files#r42735193>.
>


-- 
*#NSBcon 2015 • The "All About NServiceBus" conference is back *
*• Learn more <http://go.particular.net/l/27442/2015-08-21/5d4wzp> →*